Automattic/woocommerce-subscriptions-core
# Replace all remaining `get_post_type()`

https://github.com/Automattic/woocommerce-subscriptions-core/issues/288

## Description

- Change changelog entry per [@mattallan's suggestion](https://github.com/Automattic/woocommerce-subscriptions-core/pull/306#issuecomment-1327077739).
- Replace `get_post_type( $id )` with `WC_Data_Store::load( 'subscription' )->get_order_type( $id )`;

## How to test this PR



For `WC_Subscriptions_Product::user_can_not_delete_subscription` (`includes/class-wc-subscriptions-product.php`):
- Create a subscription product (let's call it `Subs1`).
- Purchase `Subs1`.
- Trash `Subs1`.
- Create another subscription product, `Subs2`.
- Trash `Subs2`.
- Go to the trash list (/wp-admin/edit.php?post_status=trash&post_type=product) and make sure you cannot delete permanently `Subs1` but can do so for `Subs2`.

Note: with HPOS on, `Delete Permanently` show up even for subscription products that are associated to an order. I created a ticket [here](https://github.com/Automattic/woocommerce-subscriptions-core/issues/319).

<br />

For `WCS_Admin_Post_Types::print_bulk_actions_script` (`includes/admin/class-wcs-admin-post-types.php`):
- Open the admin edit page of an active subscription.
- Look at the HTML source of that page, make sure you can find `.appendTo( "select[name='action'], select[name='action2']" );`.

Note: as of currently, we might not be able to test this with HPOS on.

<br />

For `WCS_Cached_Data_Manager::purge_delete` (`includes/class-wcs-cached-data-manager.php`):
No need to test as it's deprecated. Make sure code diff looks good.

<br />

For `WCS_Admin_Post_Types::print_bulk_actions_script` (`includes/admin/class-wcs-admin-post-types.php`):
- Make sure bulk actions show up on subscriptions admin list (`/wp-admin/edit.php?post_type=shop_subscription`).
![Screenshot 2022-12-01 at 11 59 48](https://user-images.githubusercontent.com/73803630/204969476-c9ffc0ec-07bd-4953-a7fa-5206271e8417.png)


## Product impact
